tests/sqlparser_common.rs

```diff
@@ -11748,3 +11748,16 @@ fn parse_create_table_select() {
         );
     }
 }
+
+#[test]
+fn overflow() {
+    let expr = std::iter::repeat("1")
+        .take(1000)
```

Comment:
1000 is reasonable for general usage, but 1000 won't overflow without recursive macro. Maybe we could increase it.

Reply:
I think it depends on the platform. IMO, the most important thing is that it overflows the stack [in the CI](https://github.com/blaginin/datafusion-sqlparser-rs/actions/runs/11870319597/job/33081591792)

Reply:
Sorry I didn't see Ci fail on that, I reproduce it on my laptop and 1000 seems fined :smile: